src/__tests__/interception.test.ts

```diff
@@ -0,0 +1,285 @@
+import {
+  instance as globalContainer,
+  ResolutionType
+} from "../dependency-container";
+import Frequency from "../types/frequency";
+
+// beforeResolution .resolve() tests
+test("beforeResolution interceptor gets called correctly", () => {
+  class Bar {}
+  let interceptorCalled = false;
+  globalContainer.beforeResolution(Bar, (_, resolutionType) => {
+    interceptorCalled = true;
+    expect(resolutionType).toEqual(ResolutionType.Single);
+  });
+  globalContainer.resolve(Bar);
+
+  expect(interceptorCalled).toBeTruthy();
+});
+test("beforeResolution interceptor gets called correctly", () => {
```

Comment:
This has the same test name as above. Could you rename one/both to be more specific?